### ISSUE TITLE
Stable public API

### ISSUE THREAD

**Issue by charlesdaniels** [charlesdaniels/bitshuffle#39]
A stable API to use BitShuffle as a library needs to be designed, implemented, and documented. 

As a component of this, the entire API should be tested by unit tests (which will constitute the completion of #5). 

The documentation should be generated automatically, and we can host it on [Read the Docs](https://readthedocs.org).

**Comment by jyn514** [comment]
Looks like Read the Docs [accepts markdown](https://docs.readthedocs.io/en/latest/getting_started.html)

**Comment by charlesdaniels** [comment]
Yes, I have succeeded in getting both Markdown and reST to co-exist in the same Sphinx wiki. All we would need is a build script.
